### ISSUE TITLE
Controlnet Photopea ads: is it okay?

### ISSUE THREAD

**Issue by light-and-ray** [AUTOMATIC1111/stable-diffusion-webui-extensions#255]
I found in one of the previous updates of control-net extension, they added photopea editor. It is enabled by default.

![Screenshot_20240104_232959](https://github.com/AUTOMATIC1111/stable-diffusion-webui-extensions/assets/33491867/297873e2-44c4-4508-a6fa-6420165eb788)

![Screenshot 2024-01-04 at 23-38-50 Stable Diffusion](https://github.com/AUTOMATIC1111/stable-diffusion-webui-extensions/assets/33491867/417bb8e8-68be-4fc6-b058-2d910914f558)

It looks like an awful advertisement of a viral online Photoshop clone. 👿 And I know, according to their terms of use, they collect all your data.

At least they either have to disable it by default, or it's needed to add `'ads'` and `'online'` tags into the extensions index

@Mikubill @huchenlei

**Comment by huchenlei** [comment]
The iframe only loads `photopea.com` when you click `edit`.

If you do not like the feature you can also disable it
.

**Comment by huchenlei** [comment]
![1704399618298](https://github.com/AUTOMATIC1111/stable-diffusion-webui-extensions/assets/20929282/2a927e9f-6386-41aa-92b0-745e5bac54e2)

**Comment by huchenlei** [comment]
There is an A1111 photopea extension long ago: https://github.com/yankooliveira/sd-webui-photopea-embed. I did not see people complaining about it. If you don't like it, you have the option to just not use it.

**Comment by huchenlei** [comment]
Also according README:

> - online tag is Required for any extension that connections to external server during **regular use** aside from one time downloading of assets. 
> - ads tag is Required for any extension that contains advertisements or self-advertisement **in the extension itself**.

- ControlNet extension's regular use does not depend on the photopea edit feature.
- ADs is not in the extension itself, it is from an external service provider.

**Comment by w-e-w** [comment]
in my opinion this one is in gray area

from my point of view the rule is to prevent people from blatantly using web UI as an advertising ground
the rule isn't there then people will start adding lots of different `donate` or `start the repo` buttons sprinkled in the UI
and it will be an entire mess

in my opinion adding a button that adds function that also links to a service can be recorded as a functionality as opposed to an advertisement

for instance those Civitai model downloader / integration extension, the existence of the extension itself is an advertisement of Civitai

basically my line for advertisement tag is that "if the link or button does not serve other purpose other than to advertise then it's an advertisement"

linking to Repository with the title "leave a star" is an advertisement but if it is a complicated extension that requires documentation they can add a link for documentation purposes

**Comment by w-e-w** [comment]
@huchenlei 
![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui-extensions/assets/40751091/c123e5c3-0f71-4036-a5d2-9fecf8ede333)
well did actual testing and apparently you do call to photopea for there icon

I posted this issue in Discord I want to have a different opinion on this matter but

if that icon request is in contorlnet the it will have to be tag `online`

also https://github.com/Mikubill/sd-webui-controlnet/blame/main/javascript/openpose_editor.js#L4
I think the test should be only be performed if people tryies to use is openpose-editor as opposed to on page load

**Comment by w-e-w** [comment]
> I've noticed about icon too. But he inlined it 4 hr ago: [Mikubill/sd-webui-controlnet@1679847](https://github.com/Mikubill/sd-webui-controlnet/commit/16798475b5c203cbbbf0aad234275f0d2ead710b)

ahhh ok

**Comment by akx** [comment]
👍 that the icon was inlined. It would be nice if clicking Edit `confirm()`ed that it's okay to load Photopea from Photopea.com on the first use (if it doesn't do that already).

**Comment by w-e-w** [comment]
there's a open source web page painting application called [mini-paint](https://github.com/viliusle/miniPaint)
someone even made an extension https://github.com/0Tick/a1111-mini-paint
so an alternative is to drop photopea and use mini-paint it's installed locally so no one will have issues
> aside from people that are used to using photopea

I don't know how often people edit annotated image manually
yes photopea future rich at what do users actually need?

**Comment by huchenlei** [comment]
- Confirm prompt box added: https://github.com/Mikubill/sd-webui-controlnet/pull/2411
- Openpose editor availability test delayed: https://github.com/Mikubill/sd-webui-controlnet/pull/2412

**Comment by huchenlei** [comment]
I am open to use another backend service for edit, but I am not implementing it as personally I am fine using photopea. 

Please make a PR to support other backend services in ControlNet extension repo, and I am happy to reivew and merge the change. Thanks!

**Comment by w-e-w** [comment]
from the perspective of the maintainer of the index

I don't generally don't want to get involved in the direction of an extension unless it is doing something that requires intervention

my take on this issue

I don't consider the photopea integration to be an advertisement

and after adding a popup, and no re, I think it is a good enough warning to not warrant the use of online tag

> one thing though, I think the message popup should not just show once, but every time unless a checkbox of some kind is checked
currently after if I click cancel and ok, the next time I clicked on the photopea icon it launches photopea automatic 

while yes it does encourage the use of that service, but this is not the main concern of ads tag
otherwise extensions encourage just a use of it service provided by a company
such as AWS and Civitai integrations, among others will also be classified as advertisement

I consider this issue closed unless some who wants to object further

---

by the way I just did a quick search and apparently there is an extension that integrates photopea into webui
based on readme there is controlnet integration of some kind 

for now I will pretend I haven't seen it because I don't want to test it, but if someone makes a request it can be added to the index (if it works)

---
---

form a personal perspective

I don't like photope, past experience is horrible, I don't want to use it if I can, I have disabled in controlnet, I know it's gonn break some time in the future if photpea decides to change stuff etc
..... ranting just ignore me

it should be disabled by default (opt-in not opt-out)

and best switch to a open source alternative

**Comment by w-e-w** [comment]
he answering that or not is his personal choice not concerned of this index
